packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,499 @@
+                        alpha: color.alpha ? color.alpha / 255 : 1,
+                        red: color.red ?? 0,
+                        green: color.green ?? 0,
+                        blue: color.blue ?? 0,
+                        name: color.name ?? '',
+                    };
+                });
+
+                return {
+                    id: palette.id,
+                    title: palette.name,
+                    colors,
+                };
+            })
+        );
+    }, [appBridgePalettes, appBridge]);
+
+    useEffect(() => {
+        if (blockSettings?.colorInput) {
+            setDisplayableItems(blockSettings.colorInput);
+        }
+        const currentHeight = blockSettings.customHeight ? blockSettings.heightInput : blockSettings.heightSlider;
+
+        if (colorScaleHeight !== currentHeight) {
+            setColorScaleHeight(currentHeight);
+        }
+
+        const hasColors = colorScaleBlockRef.current && blockSettings.colorInput?.length >= MINIMUM_NUMBER_OF_COLORS;
+
+        const needToCalculateWidths = blockSettings.colorInput?.find((color) => !color.width);
+
+        if (needToCalculateWidths && hasColors) {
+            const colorsWithNewWidths = calculateWidths(blockSettings.colorInput, colorScaleBlockRef, false);
+
+            setBlockSettings({
+                ...blockSettings,
+                colorInput: colorsWithNewWidths,
+            });
+        }
+    }, [blockSettings, colorScaleHeight, setBlockSettings]);
+
+    const handleResizeStop = () => {
+        draggingId.current = null;
+    };
+
+    const handleResizeStart = (event: MouseEvent, index?: number, currentColor?: ColorProps): void => {
+        if (dragStartPos) {
+            dragStartPos.current = event.clientX;
+            dragStartWidth.current = currentColor?.width;
+        }
+        draggingId.current = index;
+    };
+
+    const handleResize = (event: MouseEvent) => {
+        if (draggingId.current !== null) {
+            const colorIndex = draggingId.current;
+            if (!lastDragPos.current) {
+                lastDragPos.current = event.clientX;
+            }
+
+            if (event.clientX < lastDragPos?.current && lastDragPos.current - event.clientX >= MINIMUM_COLOR_WIDTH) {
+                lastDragPos.current = event.clientX;
+
+                let valuesChanged = false;
+
+                if (!dragStartPos.current) {
+                    return;
+                }
+
+                const movementSinceStart = dragStartPos.current - event.clientX;
+
+                const colorsBeforeCurrent = displayableItems
+                    ?.filter((_, diIndex) => colorIndex !== undefined && diIndex < colorIndex)
+                    .reverse();
+
+                const newDisplayableItems = displayableItems.map((diValue, diIndex) => {
+                    if (diValue && diIndex === colorIndex) {
+                        if (diValue.width && diValue.width >= MINIMUM_COLOR_WIDTH * 2) {
```

Comment:
Could you clarify how this could be improved? I've updated the variable names to try and make things clearer, and also  changed the first `if` to `if (colorIndex === resizedColorIndex)`.

Reply:
You have if inside if inside if for no reason. put `diValue.width && diValue.width >= MINIMUM_COLOR_WIDTH * 2` and `dragStartWidth.current ?? 0) - movementSinceStart >= MINIMUM_COLOR_WIDTH` into variables of what they represent and then simple do if(variable1 && variable2).